components/fxa-client/ios/FxAClient/FirefoxAccount.swift

```diff
@@ -46,75 +46,48 @@ public protocol PersistCallback {
-                os_log("FirefoxAccount internal state serialization failed.")
-            }
-        }
+    private init(raw: UInt64) {
+        self.raw = raw
     }
 
-    #if BROWSERID_FEATURES
-    /// Creates a `FirefoxAccount` instance from credentials obtained with the onepw FxA login flow.
-    /// This is typically used by the legacy Sync clients: new clients mainly use OAuth flows and
-    /// therefore should use `init`.
+    /// Create a `FirefoxAccount` from scratch. This is suitable for callers using the
+    /// OAuth Flow.
     /// Please note that the `FxAConfig` provided will be consumed and therefore
     /// should not be re-used.
-    open class func from(config: FxAConfig, webChannelResponse: String) throws -> FirefoxAccount {
-        return try queue.sync(execute: {
-            let handle = try FxAError.unwrap({err in
-                fxa_from_credentials(config.contentUrl, config.clientId, config.redirectUri, webChannelResponse, err)
+    public convenience init(config: FxAConfig) throws {
+        let pointer = try queue.sync(execute: {
+            return try FirefoxAccountError.unwrap({err in
+                fxa_new(config.contentUrl, config.clientId, config.redirectUri, err)
             })
-            return FirefoxAccount(raw: handle)
         })
+        self.init(raw: pointer)
     }
-    #endif
 
     /// Restore a previous instance of `FirefoxAccount` from a serialized state (obtained with `toJSON(...)`).
     open class func fromJSON(state: String) throws -> FirefoxAccount {
         return try queue.sync(execute: {
-            let handle = try FxAError.unwrap({ err in fxa_from_json(state, err) })
+            let handle = try FirefoxAccountError.unwrap({ err in fxa_from_json(state, err) })
             return FirefoxAccount(raw: handle)
         })
     }
 
-    /// Create a `FirefoxAccount` from scratch. This is suitable for callers using the
-    /// OAuth Flow.
-    /// Please note that the `FxAConfig` provided will be consumed and therefore
-    /// should not be re-used.
-    public convenience init(config: FxAConfig) throws {
-        let pointer = try queue.sync(execute: {
-            return try FxAError.unwrap({err in
-                fxa_new(config.contentUrl, config.clientId, config.redirectUri, err)
-            })
-        })
-        self.init(raw: pointer)
-    }
-
-    override func cleanup(pointer: UInt64) {
-        queue.sync(execute: {
-            try! FxAError.unwrap({err in
-                // Is this the right thing to do? We should only hit an error here
-                // for panics and handle misuse, both inidicate bugs in our code
-                // (the first in the rust code, the 2nd in this swift wrapper).
-                fxa_free(pointer, err)
+    deinit {
+        if self.raw != 0 {
+            queue.sync(execute: {
+                try! FirefoxAccountError.unwrap({err in
+                    // Is this the right thing to do? We should only hit an error here
```

Comment:
I think I wrote this comment, it's not clear I'm talking about `try!` here, could you change this to say "Is `try!` the right thing to do?" instead